cfn-resources/stream-connection/cmd/resource/mappings.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+func GetStreamConnectionModel(streamsConn *admin.StreamsConnection, currentModel *Model) *Model {
+	model := &Model{}
+
+	if currentModel != nil {
+		model = currentModel
+	}
+
+	model.ConnectionName = streamsConn.Name
+	model.Type = streamsConn.Type
+	model.ClusterName = streamsConn.ClusterName
+	model.BootstrapServers = streamsConn.BootstrapServers
+
+	model.DbRoleToExecute = NewModelDBRoleToExecute(streamsConn.DbRoleToExecute)
+
+	model.Authentication = NewModelAuthentication(streamsConn.Authentication)
+
+	model.Security = NewModelSecurity(streamsConn.Security)
+
+	if streamsConn.Config != nil {
+		model.Config = *streamsConn.Config
+	}
+
+	return model
+}
+
+func NewModelDBRoleToExecute(dbRole *admin.DBRoleToExecute) *DBRoleToExecute {
+	if dbRole != nil {
+		return &DBRoleToExecute{
+			Role: dbRole.Role,
+			Type: dbRole.Type,
+		}
+	}
+	return nil
```

Comment:
[nit] I think is more readable (personal preference). This also applies to all the other similar functions.
```suggestion
	if dbRole == nil {
	   return nil
	  }
		return &DBRoleToExecute{
			Role: dbRole.Role,
			Type: dbRole.Type,
		}
```

Reply:
agreed with @andreaangiolillo , it's more idiomatic in Go to reduce nesting

Reply:
updated